bindings/pyroot/pythonizations/python/ROOT/_pythonization/_tmva/_batchgenerator.py

```diff
@@ -491,6 +516,9 @@ def __iter__(self):
         return self
 
     def __next__(self):
+        if not hasattr(self, '_callable'):
+            self._callable = self.__call__()
```

Comment:
Is there any situation where `self._callable` is something other than `self.__call__()`? Otherwise, this condition can just be removed and just use `self.__call__()` directly (aka `self()`)